core/src/Streamly/Internal/Data/MutArray/Generic.hs

```diff
@@ -479,9 +476,9 @@ snoc = snocWith (* 2)
 {-# INLINE uninit #-}
 uninit :: MonadIO m => MutArray a -> Int -> m (MutArray a)
 uninit arr@MutArray{..} len =
-    if arrStart + arrLen + len <= arrTrueLen
-    then return $ arr {arrLen = arrLen + len}
-    else realloc (arrLen + len) arr
+    if arrEnd + len <= arrCapacity
```

Comment:
Looks like the old code was wrong. If the meaning of arrTrueLen was length instead of index.

Reply:
Why is the old code wrong? I did not understand.